src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -0,0 +1,271 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.xpath.AbstractNode;
+import com.puppycrawl.tools.checkstyle.xpath.RootNode;
+import net.sf.saxon.Configuration;
+import net.sf.saxon.om.Item;
+import net.sf.saxon.sxpath.XPathDynamicContext;
+import net.sf.saxon.sxpath.XPathEvaluator;
+import net.sf.saxon.sxpath.XPathExpression;
+import net.sf.saxon.trans.XPathException;
+
+/**
+ * <p>
+ * Evaluates Xpath query and report violation on all matching AST nodes. This check allows
+ * user to implement custom checks using Xpath. If Xpath query is not specified explicitly,
+ * then the check does nothing.
+ * </p>
+ * <p>
+ * Please read more about Xpath syntax at
+ * <a href="https://www.saxonica.com/html/documentation/expressions/">Xpath Syntax</a>.
+ * Information regarding Xpath functions can be found at
+ * <a href="https://www.saxonica.com/html/documentation/functions/fn/">XSLT/XPath Reference</a>.
+ * Note, that <b>@text</b> attribute can used only with token types that are listed in
```

Comment:
@timurt , please recheck  rendering of ` <b>@text</b>` you might need to escape `@`

Reply:
yeah, I have generated sites and it looks fine as javadoc and as checkstyle site as well, do you want me to upload this htmls to pages?

Reply:
looks good
![image](https://user-images.githubusercontent.com/812984/101027148-9a63d880-352c-11eb-9948-79d7eed9f2cf.png)